File: app/src/main/java/com/cdd/smartsurvey/HeaderSurveyMasterActivity.java
package com.cdd.smartsurvey;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.content.ContextCompat;
import androidx.core.content.res.ResourcesCompat;

import com.cdd.smartsurvey.sqlite.DatabaseHelper;
import com.cdd.smartsurvey.sqlite.model.SurveyMetric;
import com.cdd.smartsurvey.utils.ImageUtil;

import java.util.ArrayList;
import java.util.List;

public class HeaderSurveyMasterActivity extends AppCompatActivity {

    public String taggroup;
    public String imgicon;
    public String headervalue;
    public String metricvalue;
    public SurveyMetric item;

    public ImageView imgIcon;
    public TextView HeaderValue;
    public TextView MetricValue;
    public Button btnBack;
    public LinearLayout LinearMaster;

    @SuppressLint("SourceLockedOrientationActivity")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_survey_metric_master);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

        btnBack = (Button) findViewById(R.id.btnBack);
        btnBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });

        Intent intent = getIntent();
        taggroup = intent.getStringExtra("taggroup");
        imgicon = intent.getStringExtra("imgicon");
        headervalue = intent.getStringExtra("headervalue");
        metricvalue = intent.getStringExtra("metricvalue");

        imgIcon = (ImageView) findViewById(R.id.imgIcon);
        HeaderValue = (TextView) findViewById(R.id.txtHeaderValue);
        MetricValue = (TextView) findViewById(R.id.txtMetricValue);
        LinearMaster = (LinearLayout) findViewById(R.id.linearBody);

        imgIcon.setImageBitmap(ImageUtil.convert(imgicon));
        HeaderValue.setText(headervalue);
        MetricValue.setText("ประกอบด้วย " + metricvalue);

        DatabaseHelper db;

        final List<SurveyMetric> surveyMetricsList = new ArrayList<>();
        db = new DatabaseHelper(this);
        surveyMetricsList.addAll(db.getAllSurveyMetrics(taggroup));

        for (int i = 0; i < surveyMetricsList.size(); i++) {
            item = surveyMetricsList.get(i);
            LinearMaster.addView(setFrameLayout(item.getTbl_survey_group_masterid(), item.getMetric_no(), item.getMetric_display(), imgicon, item.getMetric_description()));
        }
    }

    public TextView setTextView(final String surveyGroupID, final String metricNo, final String metricDisplay, final String intValue, final String metricDesc) {

        TextView TextMaster = new TextView(getApplicationContext());
        ConstraintLayout.LayoutParams paramsText = new ConstraintLayout.LayoutParams(
                ConstraintLayout.LayoutParams.WRAP_CONTENT,
                ConstraintLayout.LayoutParams.WRAP_CONTENT
        );
        TextMaster.setLayoutParams(paramsText);
        TextMaster.setTypeface(ResourcesCompat.getFont(getApplicationContext(), R.font.thsarabun_bold));
        TextMaster.setGravity(Gravity.CENTER_VERTICAL);
        TextMaster.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.colorPrimary));
        TextMaster.setText(metricDisplay);
        TextMaster.setTextSize(20);
        ViewGroup.MarginLayoutParams params = (ViewGroup.MarginLayoutParams) TextMaster.getLayoutParams();
        params.setMargins((int) (20 * getResources().getSystem().getDisplayMetrics().density),
                (int) (20 * getResources().getSystem().getDisplayMetrics().density),
                (int) (20 * getResources().getSystem().getDisplayMetrics().density), 0);

        TextMaster.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), SurveySubMasterActivity.class);
                intent.putExtra(SurveySubMasterActivity.EXTRA_SurveyGroupID, surveyGroupID);
                intent.putExtra(SurveySubMasterActivity.EXTRA_MetricNo, metricNo);
                intent.putExtra(SurveySubMasterActivity.EXTRA_MetricDisplay, metricDisplay);
                intent.putExtra(SurveySubMasterActivity.EXTRA_Image, intValue);
                intent.putExtra(SurveySubMasterActivity.EXTRA_MetricDesc, metricDesc);
                startActivity(intent);
            }
        });
        return TextMaster;
    }

    public FrameLayout setFrameLayout(final String surveyGroupID, final String metricNo, final String metricDisplay, final String intValue, final String metricDesc) {

        FrameLayout FrameMaster = new FrameLayout(getApplicationContext());
        FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.MATCH_PARENT,
                FrameLayout.LayoutParams.MATCH_PARENT
        );
        FrameMaster.setLayoutParams(lp);
        FrameMaster.setBackground(getResources().getDrawable(R.drawable.bordergroup));
        FrameMaster.addView(setTextView(surveyGroupID, metricNo, metricDisplay, intValue, metricDesc));
        ViewGroup.MarginLayoutParams params = (ViewGroup.MarginLayoutParams) FrameMaster.getLayoutParams();
        params.setMargins((int) (10 * getResources().getSystem().getDisplayMetrics().density), 0,
                (int) (10 * getResources().getSystem().getDisplayMetrics().density), 0);
        FrameMaster.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), SurveySubMasterActivity.class);
                intent.putExtra(SurveySubMasterActivity.EXTRA_SurveyGroupID, surveyGroupID);
                intent.putExtra(SurveySubMasterActivity.EXTRA_MetricNo, metricNo);
                intent.putExtra(SurveySubMasterActivity.EXTRA_MetricDisplay, metricDisplay);
                intent.putExtra(SurveySubMasterActivity.EXTRA_Image, intValue);
                intent.putExtra(SurveySubMasterActivity.EXTRA_MetricDesc, metricDesc);
                startActivity(intent);
            }
        });
        return FrameMaster;
    }
}

File: app/src/main/java/com/cdd/smartsurvey/HeaderSurveyMasterActivity.kt
package com.cdd.smartsurvey

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.cdd.smartsurvey.adapter.SurveyMetricRecyclerViewAdapter
import com.cdd.smartsurvey.sqlite.DatabaseHelper
import com.cdd.smartsurvey.sqlite.model.SurveyGroup
import com.cdd.smartsurvey.sqlite.model.SurveyMetric
import com.cdd.smartsurvey.utils.ImageUtil
import kotlinx.android.synthetic.main.activity_survey_metric_master.*

class HeaderSurveyMasterActivity : AppCompatActivity() {
    lateinit var surveyGroup: SurveyGroup
    lateinit var surveyMetricsList: List<SurveyMetric>
    var familyIndex = 0
    var item: SurveyMetric? = null

    @SuppressLint("SourceLockedOrientationActivity")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_survey_metric_master)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        btnBack.setOnClickListener { onBackPressed() }
        val surveyGroupID = intent.getIntExtra(GlobalValue.EXTRA_SURVEY_GROUP_ID, 0)
        familyIndex = intent.getIntExtra(GlobalValue.EXTRA_FAMILY_INDEX, 0)


        loadSurveyListGroup(surveyGroupID)
    }

    private fun loadSurveyListGroup(groupID: Int) {
        val db = DatabaseHelper(this)
        surveyGroup = db.getSurveyGroup(groupID)!!
        imgIcon.setImageBitmap(ImageUtil.convert(surveyGroup.groupImage))
        txtHeaderValue.text = surveyGroup.groupName
        txtMetricValue.text = "ประกอบด้วย ${surveyGroup.groupMetric}"
        surveyMetricsList = db.getAllSurveyMetrics(groupID.toString())
        listViewSurveyMetric.layoutManager = LinearLayoutManager(this)
        listViewSurveyMetric.setHasFixedSize(true)
        listViewSurveyMetric.adapter = SurveyMetricRecyclerViewAdapter(surveyMetricsList) { view, i, surveyMetric ->
            val intent = Intent(applicationContext, SurveySubMasterActivity::class.java).apply {
                putExtra(GlobalValue.EXTRA_FAMILY_INDEX, familyIndex)
                putExtra(GlobalValue.EXTRA_SURVEY_GROUP_ID, surveyGroup.id)
                putExtra(GlobalValue.EXTRA_SURVEY_METRIC_INDEX, i)
            }
            startActivityForResult(intent, 100)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == 100) {
            when (resultCode) {
                Activity.RESULT_OK -> {
                    val surveyGroupID = data?.getIntExtra(GlobalValue.EXTRA_SURVEY_GROUP_ID, 0)
                    if (surveyGroupID != 0) {
                        loadSurveyListGroup(surveyGroupID!!)
                    }
                }
                GlobalValue.RESULT_FINISH -> {
                    val resultIntent = Intent()
                    setResult(GlobalValue.RESULT_FINISH, resultIntent)
                    finish()
                }
            }

        }

        super.onActivityResult(requestCode, resultCode, data)
    }
}